core/runtime/src/main/java/io/quarkus/runtime/logging/LoggingSetupRecorder.java

```diff
@@ -687,6 +687,9 @@ private static Handler configureSyslogHandler(final SyslogConfig config, final E
             handler.setTruncate(config.truncate);
             handler.setUseCountingFraming(config.useCountingFraming);
             handler.setLevel(config.level);
+            if (config.maxLength.isPresent()) {
+                handler.setMaxLength(config.maxLength.get().asBigInteger().intValue());
```

Comment:
There is actually an implicit limit of 2GB here; if the user exceeds it then it will silently wrap to a negative (invalid) or unexpected (smaller) value. We should either validate the value or else clamp it into range.

It is probably a good idea to set a reasonable lower limit as well.

Reply:
PR updated as per suggestions